python-avd/pyavd/_eos_designs/shared_utils/routing.py

```diff
@@ -135,8 +135,11 @@ def bgp_as(self: SharedUtilsProtocol) -> str | None:
                 raise AristaAvdInvalidInputsError(msg)
             return bgp_as_range_expanded[self.id - 1]
         except IndexError as exc:
-            msg = f"Unable to allocate BGP AS: bgp_as range is too small ({len(bgp_as_range_expanded)}) for the id of the device"
-            raise AristaAvdError(msg) from exc
+            msg = (
+                f"Unable to allocate BGP AS: bgp_as range '{self.node_config.bgp_as}' is too small ({len(bgp_as_range_expanded)}) for the id "
+                f"'{self.id}' of the device"
```

Comment:
Looking at the error messages we need to clean this.

```suggestion
                f"'{self.id}'."
```